lightly/api/api_workflow_upload_dataset.py

```diff
@@ -73,16 +73,28 @@ def upload_dataset(self, input: Union[str, LightlyDataset], max_workers: int = 8
         if verbose:
             print(f'Uploading images (with {max_workers} workers).', flush=True)
 
-        pbar = tqdm.tqdm(unit='imgs', total=len(dataset))
-        tqdm_lock = tqdm.tqdm.get_lock()
 
         # calculate the files size more efficiently
         lightly_utils.image_processing.metadata._size_in_bytes = lambda img: 0
 
+        # get the filenames of the samples already on the server
+        self.samples_api: SamplesApi
+        samples: List[SampleData] = self.samples_api.get_samples_by_dataset_id(dataset_id=self.dataset_id)
+        filenames = [sample.file_name for sample in samples]
+        if len(filenames) > 0:
+            print(f"Found {len(filenames)} images already on the server, they are skipped during the upload.")
+        filenames_set = set(filenames)
+
+        pbar = tqdm.tqdm(unit='imgs', total=len(dataset)-len(filenames))
+        tqdm_lock = tqdm.tqdm.get_lock()
+
         # define lambda function for concurrent upload
         def lambda_(i):
             # load image
             image, label, filename = dataset[i]
+            if filename in filenames_set:
```

Comment:
can you add a single line comment here? something like `# file was already uploaded`

Reply:
Implemented it.